### PR TITLE
REF: simplify _sanitize_column

### PR DESCRIPTION
**Pull request by jbrockmendel**
It gets called indirectly from `iloc._setitem_with_indexer`, which we're trying to simplify

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -3210,15 +3212,47 @@ def _setitem_frame(self, key, value):
         self._check_setitem_copy()
         self._where(-key, value, inplace=True)
 
+    def _set_item_frame_value(self, key, value: "DataFrame"):
```
Comment (jreback) [inline pandas/core/frame.py:3215]: add return type (None)

File: pandas/core/frame.py [hunk 2]
```diff
@@ -9552,3 +9536,34 @@ def _from_nested_dict(data) -> collections.defaultdict:
         for col, v in s.items():
             new_data[col][index] = v
     return new_data
+
+
+def _reindex_for_setitem(value, index: Index):
+    # reindex if necessary
+
+    if value.index.equals(index) or not len(index):
+        value = value._values.copy()
```
Comment (jreback) [inline pandas/core/frame.py:9545]: can you just return here?

File: pandas/core/frame.py [hunk 3]
```diff
@@ -9552,3 +9536,34 @@ def _from_nested_dict(data) -> collections.defaultdict:
...
+    else:
+
+        # GH#4107
+        try:
```
Comment (jreback) [inline pandas/core/frame.py:9549]: avoids nesting this

File: pandas/core/frame.py [hunk 4]
```diff
@@ -9552,3 +9536,34 @@ def _from_nested_dict(data) -> collections.defaultdict:
...
+                # duplicate axis
+                raise err
+
+            raise TypeError(
+                "incompatible index of inserted column with frame index"
+            ) from err
+    return value
+
+
+def _maybe_atleast_2d(value):
```
Comment (jreback) [inline pandas/core/frame.py:9563]: can likley use elsewhere

Comment (jreback) [reply pandas/core/frame.py:9563]: could put in core/dtypes/cast

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/core/frame.py:9563]: ill take a look; most likely place it would be used is in internals

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks, yeah we have a lot of built up code over the years. +1 on organizaing / simplifying.

**Pull request merged**